wormhole_attester/client/src/main.rs

```diff
@@ -662,19 +667,42 @@ async fn attestation_job(args: AttestationJobArgs) -> Result<(), ErrBoxSend> {
 
         let wh_msg_id = message_q_mtx.lock().await.get_account()?.id;
 
-        let tx_res: Result<_, ErrBoxSend> = gen_attest_tx(
+        let tx = gen_attest_tx(
             p2w_addr,
             &config,
             &payer,
             wh_msg_id,
             symbols.as_slice(),
             latest_blockhash,
-        );
+            rate_limit_interval_secs,
+        )?;
+
+        // Detect rate limiting error early
```

Comment:
I am a bit hesitant to this change because the RPC latency (which seems to be around 1s) might actually make this check worthless and even reduce the actual attestation rate hitting the network (regardless of the result). 

So 2 unfortunate consequences are:
1. You decide not to attest because the simulation fails, but if you continue the instruction and it actually sends the transaction it might not get rate limited.
2. You decide to attest and then it gets rate limited when it's sent for real.

I think it's better to send the tx and only capture the log of its result instead.

The slightly more nuance here is that rpc performs simulation before the actual tx send. I assume that when the contract returns error then the transaction will get reverted, right?
So they do the simulation and the above scenario's can still happen that actually make our performance a bit un-reliable. 
My suggestion is that you do not fail return an error when it is rate limited and just keep sending the msg to the log (either it is attesting or not). Then you can also have 2 metrics instead of 1 for sent messages. one for sent tx and one for tx es that really attest.

Reply:
Thanks for noticing this. I thought about this and for some reason I couldn't find a way to get the logs from an error in `solana_client` :(

Reply:
Because in that case, We would only need to do a `map_err` and fish the expected log line out of the error type.

Reply:
Come to think of it, the right thing to do might be using a custom error code instead of a log line, I think I'll do that instead.

Reply:
Hmm, Maybe, but I don't like _not failing_ the transaction very much, because preflight failures are guaranteed to make this free

Reply:
But our concern is not much the fees, they are essentially free right?

Reply:
I think you can disable preflight simulation (I know how to do it on javascript).

Reply:
I think Guillermo is right. I'll keep thinking about this for a while and settle for a preflight-less tx. This is probably the only way to get 100% precise rate limiting of the wormhole cross-txs

Reply:
Okay, I think I came up with the right compromise. We use an error code, look for it in failing tx, finish attestation job if it's detected, pass the error through otherwise. This means the attester will **never** breach the desired rate limit, and at absolute worst pay for the failing tx if  the rare race condition is met.

Reply:
Tilt just went green locally.